hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1064,8 +1065,30 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	switch hcluster.Spec.Platform.Type {
 	case hyperv1.AWSPlatform:
 		if err := r.reconcileAWSOIDCDocuments(ctx, log, hcluster, hcp); err != nil {
+			meta.SetStatusCondition(&hcluster.Status.Conditions, metav1.Condition{
+				Type:               string(hyperv1.OIDCConfigurationInvalid),
```

Comment:
Thoughts about `OIDCConfigurationInvalid` being the reason for an `OperatorReadyConditionType  = false`?

Reply:
Also thoughts about spending a s3 put in main to validate and fail earlier the operator otherwise?